src/python/pants/option/option_types.py

```diff
@@ -0,0 +1,481 @@
+    # short as possible to avoid having to repeat the param in each of the 3 `__new__` specs that
+    # each subclass must provide.
+    # If you need additional information, this class follows the "Builder" pattern. Use the
+    # "builder" methods to attach additional information. (E.g. see `advanced` method)
+    # NB: We define `__new__` rather than `__init__` because otherwise subclasses would have to
+    # define 3 `__new__`s AND `__init__`.
+    def __new__(
+        cls,
+        *flag_names: str,
+        default: Any,
+        option_type: Type | None = None,
+        help: str,
+    ):
+        self = super().__new__(cls)
+        self.flag_names = flag_names
+        if option_type is None:
+            option_type = cls.option_type
+
+        self.kwargs = dict(
+            type=option_type,
+            default=default,
+            help=help,
+        )
+        return self
+
+    @overload
+    def __get__(self, obj: None, objtype: Any) -> OptionBase[PropType]:
+        ...
+
+    @overload
+    def __get__(self, obj: object, objtype: Any) -> PropType:
+        ...
+
+    def __get__(self, obj, objtype):
+        if obj is None:
+            return self
+        long_name = self.flag_names[-1]
+        option_value = getattr(obj.options, long_name[2:].replace("-", "_"))
+        if option_value is None:
+            return None
+        return self._convert_(option_value)
+
+    # Subclasses can override if necessary
+    def _convert_(self, val: Any) -> PropType:
+        return cast("PropType", self.kwargs["type"](val))
+
+    # ===== "Builder" Methods =====
+    def advanced(self) -> OptionBase[PropType]:
+        self.kwargs["advanced"] = True
+        return self
+
+    def from_file(self) -> OptionBase[PropType]:
+        self.kwargs["fromfile"] = True
+        return self
+
+    def metavar(self, metavar: str) -> OptionBase[PropType]:
+        self.kwargs["metavar"] = metavar
+        return self
+
+    def mutually_exclusive_group(self, mutually_exclusive_group: str) -> OptionBase[PropType]:
+        self.kwargs["mutually_exclusive_group"] = mutually_exclusive_group
+        return self
+
+    def default_help_repr(self, default_help_repr: str) -> OptionBase[PropType]:
+        self.kwargs["default_help_repr"] = default_help_repr
+        return self
+
+    def will_be_removed(self, *, version: str, hint: str) -> OptionBase[PropType]:
+        self.kwargs["removal_version"] = version
+        self.kwargs["removal_hint"] = hint
+        return self
+
+    # TODO: These only seem relevant on global options?
+    def daemoned(self) -> OptionBase[PropType]:
+        self.kwargs["daemon"] = True
+        return self
+
+    def non_fingerprinted(self) -> OptionBase[PropType]:
+        self.kwargs["fingerprint"] = False
+        return self
+
+
+class ListOptionBase(OptionBase["tuple[ListMemberType, ...]"], Generic[ListMemberType]):
```

Comment:
With Target API, we call it `Sequence` rather than `List` because:

1. BUILD files let you use any sequence (really any iterable), including list, tuple, set
2. We store it as a tuple, not a list

What do you think about renaming this to be Sequence for parity? I honestly don't know what's better. A pro of renaming is parity with Target API, so more predictable; but also it really is a "List Option"...we'd be lying to say it's any sequence. I suspect you're right to stick with `List`.

Reply:
So I think the issue is the ambiguity of "list". I see it as an english list (like "checklist") rather than `list`. Maybe we can thesaurus this? But also meh.

Reply:
I don't think `list` is ambiguous to a Python developer. Imo it's either `list` or `sequence`. Pretty sure list is the better approach, given the nature of the subsystem API and this page: https://www.pantsbuild.org/docs/options